### PR TITLE
Add org.cznic.Datovka

### PR DESCRIPTION
**Pull request by Sesivany**

### CODE REVIEW

**Comment by AsciiWolf** [comment]
bot, build org.cznic.Datovka

**Comment by AsciiWolf** [comment]
Build succeeded.

**Comment by Sesivany** [comment]
I'm planning to approach them once it's available on Flathub and share maintainership with them and eventually hand it over to them completely. If they are not interested, I can maintain it myself.

**Comment by barthalion** [comment]
Can you contact them beforehand? While it's fine license-wise, we had at least once unpleasant situation caused by not contacting the developer.

**Comment by Sesivany** [comment]
I contacted the upstream developers and they're fine with me maintaining the app on Flathub and pointing to them as app developers, but they're not currently interested in maintaining it themselves and they would like me to clearly state in the description that it's not an official flatpak maintained by them, so that it doesn't pose any liability and responsibility for the flatpak on them.
Are there any guidelines how to do? Or should I just patch the AppStream metadata to add something like "This is not an official flatpak maintained by CZ.NIC"?

**Comment by nedrichards** [comment]
Patch the appstream data to do that. (we do this for a number of extra-data apps). It'd be good if there was a more structured way to indicate this, there's been some discussion on that but nothing concrete emerging as of yet.

File: org.cznic.Datovka.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+app-id: org.cznic.Datovka
+runtime: org.kde.Platform
+sdk: org.kde.Sdk
+runtime-version: 5.11
```
Comment (TingPing) [inline org.cznic.Datovka.yaml]:
```suggestion
runtime-version: '5.11'
```

File: org.cznic.Datovka.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+command: datovka
+rename-desktop-file: datovka.desktop
```
Comment (TingPing) [inline org.cznic.Datovka.yaml]: Can you try to get upstream to rename these files at least? The renaming is per the desktop/appstream spec.

File: org.cznic.Datovka.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+rename-appdata-file: datovka.appdata.xml
+rename-icon: datovka
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --share=network
+  - --talk-name=org.freedesktop.Notifications
+  - --filesystem=xdg-config/kdeglobals:ro
```
Comment (TingPing) [inline org.cznic.Datovka.yaml]:
```suggestion
```

File: org.cznic.Datovka.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --talk-name=ca.desrt.dconf
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
+
+modules:
+  - name: libxml2
+    buildsystem: autotools
```
Comment (TingPing) [inline org.cznic.Datovka.yaml]:
```suggestion
```

File: org.cznic.Datovka.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    config-opts:
+        - --without-python
+    sources:
+        - type: archive
+          url: https://gitlab.gnome.org/GNOME/libxml2/-/archive/v2.9.8/libxml2-v2.9.8.tar.bz2
+          sha256: 75f60530c4ff9717930ea28abcf3dfc4b213ac7617884f37637d069b01032b8a
+
+  - name: libisds
+    buildsystem: autotools
```
Comment (TingPing) [inline org.cznic.Datovka.yaml]:
```suggestion
```

File: org.cznic.Datovka.yaml (hunk 2 above)
Comment (Sesivany) [reply org.cznic.Datovka.yaml]: I'll talk to the upstream developers about it, it definitely makes sense to have the same app ID in flatpak and packages distributed in distros, but let's not stall the review on this.

**Comment by Sesivany** [comment]
I've found one glitch. They use QDir::homePath() for config and database directory instead of QStandardPath::ConfigLocation. So it's pretty much hardcoded in ~/.gui and settings are not persistent if the app doesn't have an access to the whole /home on the host.
So I'd either have to patch it or enable the access.

**Comment by AsciiWolf** [comment]
Making the directory persistent with `--persist=.gui` does not work?

**Comment by Sesivany** [comment]
> Making the directory persistent with `--persist=.gui` does not work?

I've got a patch for it, will update the manifest soon. Now I'm also trying to figure out how to run it with TMPDIR=$XDG_CACHE_HOME, so that it doesn't try to save files that need to be opened in another app in /tmp.

**Comment by Sesivany** [comment]
I've patched the config folder location and added a script that runs the app binary with TMPDIR=$XDG_CACHE_HOME. Now the app works nicely and still in the sandbox.

**Comment by nedrichards** [comment]
bot, build org.cznic.Datovka

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.cznic.Datovka

**Pull request closed without merging**